nsxt/lb_rule_utils.go

```diff
@@ -0,0 +1,290 @@
+/* Copyright © 2018 VMware, Inc. All Rights Reserved.
+   SPDX-License-Identifier: MPL-2.0 */
+
+package nsxt
+
+import (
+	"fmt"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+	api "github.com/vmware/go-vmware-nsxt"
+	"github.com/vmware/go-vmware-nsxt/loadbalancer"
+	"log"
+	"net/http"
+)
+
+func getLbRuleInverseSchema() *schema.Schema {
+	return &schema.Schema{
+		Type:        schema.TypeBool,
+		Description: "Whether to reverse match result of this condition",
+		Optional:    true,
+		Default:     false,
+	}
+}
+
+func getLbRuleCaseSensitiveSchema() *schema.Schema {
+	return &schema.Schema{
+		Type:        schema.TypeBool,
+		Description: "If true, case is significant in condition matching",
+		Optional:    true,
+		Default:     true,
+	}
+}
+
+func getLbRuleMatchTypeSchema() *schema.Schema {
+	return &schema.Schema{
+		Type:         schema.TypeString,
+		Description:  "Match type (STARTS_WITH, ENDS_WITH, EQUALS, CONTAINS, REGEX)",
+		ValidateFunc: validation.StringInSlice([]string{"STARTS_WITH", "ENDS_WITH", "EQUALS", "CONTAINS", "REGEX"}, false),
+		Required:     true,
+	}
+}
+
+func getLbRuleHTTPRequestBodyConditionSchema() *schema.Schema {
+	return &schema.Schema{
+		Type:        schema.TypeSet,
+		Description: "Rule condition based on http request body",
+		Optional:    true,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"inverse": getLbRuleInverseSchema(),
+				"value": &schema.Schema{
+					Type:     schema.TypeString,
+					Required: true,
+				},
+				"case_sensitive": getLbRuleCaseSensitiveSchema(),
+				"match_type":     getLbRuleMatchTypeSchema(),
+			},
+		},
+	}
+}
+
+func getLbRuleHTTPHeaderConditionSchema() *schema.Schema {
+	return &schema.Schema{
+		Type:        schema.TypeSet,
+		Description: "Rule condition based on http header",
+		Optional:    true,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"inverse": getLbRuleInverseSchema(),
+				"name": &schema.Schema{
+					Type:     schema.TypeString,
+					Required: true,
+				},
+				"value": &schema.Schema{
+					Type:     schema.TypeString,
+					Required: true,
+				},
+				"case_sensitive": getLbRuleCaseSensitiveSchema(),
+				"match_type":     getLbRuleMatchTypeSchema(),
+			},
+		},
+	}
+}
+
+func getLbRuleHTTPRequestMethodConditionSchema() *schema.Schema {
+	return &schema.Schema{
```

Comment:
Max 1 entry?

Reply:
Multiples are allowed (for example, not PUT and not POST)